src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx

```diff
@@ -11,31 +13,116 @@ interface Props {
 }
 
 export function FilledField({ order, sellAmount, buyAmount }: Props) {
-  const { inputToken, outputToken } = order
+  const {
+    inputToken,
+    outputToken,
+    filledPercentage,
+    fullyFilled,
+    kind,
+    sellToken,
+    buyToken,
+    feeAmount,
+    executedBuyAmount,
+    executedSellAmount,
+    executedFeeAmount,
+    filledAmount,
+  } = order
 
-  const filledPercentage = useMemo(() => {
-    if (!order || !order.filledPercentage) {
+  const touched = filledPercentage?.gt(0)
+
+  let mainToken
+  let mainAddress
+  let mainAmount
+  let swappedToken
+  let swappedAddress
+  let swappedAmount
+  let action
+
+  let filledAmountWithFee, swappedAmountWithFee
+  if (kind === OrderKind.SELL) {
```

Comment:
Should be moved to a function

Reply:
Its based on this https://github.com/cowprotocol/explorer/blob/develop/src/components/orders/FilledProgress/index.tsx so I would keep it for now and then later we can refactor this in both places.